File: 1236-n-th-tribonacci-number/n-th-tribonacci-number.c


int tribonacci(int n){
    int a = 0, b = 1, c=1;
    int tmp;
    for(int i =3; i<=n; i++){
        tmp = a + b +c;
        a = b;
        b = c;
        c = tmp;
    }
    if (n==0){
        return 0;
    }
    return c;
}

File: 1236-n-th-tribonacci-number/n-th-tribonacci-number.py
class Solution:
    def tribonacci(self, n: int) -> int:
        a = 0
        b = 1
        c = 1
        if n ==0:
            return 0
        elif n== 1 or n ==2:
            return 1
        else:
            for i in range(3,n+1):
                tmp = a + b + c
                a = b
                b = c
                c = tmp
            return c
